src/AlgebraicGeometry/Schemes/PrincipalOpenSubset/Objects/Attributes.jl

```diff
@@ -18,7 +18,7 @@ function complement_equation(U::PrincipalOpenSubset)
   if !isdefined(U, :h)
     U.h = prod(OO(U).(U.f); init=one(OO(U)))
   end
-  return U.h::elem_type(OO(ambient_scheme(U)))
+  return U.h  # ::elem_type(OO(ambient_scheme(U)))
```

Comment:
This makes me curious, though. Did this cause problems? Or is it just that with the removal of the type of the ambient scheme this is not type stable anyway?

Reply:
Actually I wanted to discuss this with you: if the goal here is to enforce an invariant and catch bugs, we can of course keep it. If the goal was ensuring type stability, then this type assertion does not provide it, and might as well be removed.

Reply:
The goal was type assertion. I think it can go, then.

Reply:
I suppose this does not help with type stability, because this is computed at runtime?